From f3421c81d7ad455b31190cc6cd4068f3c4de6378 Mon Sep 17 00:00:00 2001
From: yangyingchao <yang.yingchao@qq.com>
Date: Fri, 9 Aug 2024 20:49:31 +0800
Subject: [PATCH] Prepend ${DESTDIR} when creating hyprland symbolic link.

```
 * ACCESS DENIED:  symlink:            /usr/bin/hyprland
CMake Error: failed to create symbolic link '/usr/bin/hyprland': Permission denied

 * ----------------------- SANDBOX ACCESS VIOLATION SUMMARY -----------------------
 * LOG FILE: "/var/tmp/portage/gui-wm/hyprland-0.42.0-r1/temp/sandbox.log"
 *
VERSION 1.0
FORMAT: F - Function called
FORMAT: S - Access Status
FORMAT: P - Path as passed to function
FORMAT: A - Absolute Path (not canonical)
FORMAT: R - Canonical Path
FORMAT: C - Command Line

F: symlink
S: deny
P: /usr/bin/hyprland
A: /usr/bin/hyprland
R: /usr/bin/hyprland
C: /usr/bin/cmake -E create_symlink /usr/bin/Hyprland /usr/bin/hyprland
 * --------------------------------------------------------------------------------

```
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc8eafd5..bd607189 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -329,7 +329,7 @@ install(
   CODE "execute_process( \
         COMMAND ${CMAKE_COMMAND} -E create_symlink \
         ${CMAKE_INSTALL_FULL_BINDIR}/Hyprland \
-        ${CMAKE_INSTALL_FULL_BINDIR}/hyprland
+        \"\$ENV{DESTDIR}${CMAKE_INSTALL_FULL_BINDIR}/hyprland\" \
         )")
 
 # session file
-- 
2.44.2

